staging/src/k8s.io/client-go/dynamic/scheme.go

```diff
@@ -106,3 +100,24 @@ func (t unstructuredTyper) ObjectKinds(obj runtime.Object) ([]schema.GroupVersio
 func (t unstructuredTyper) Recognizes(gvk schema.GroupVersionKind) bool {
 	return true
 }
+
+// The dynamic client formerly hardcoded object marshaling and unmarshaling and would allow
+// Unstructured objects that were missing apiVersion and/or kind in requests.
```

Comment:
allow in serializing requests or in deserializing responses?

Reply:
Serializing requests. Directly encoding with `(unstructuredJSONScheme).Encode` is effectively a pass-through to `json.Marshal`. With this PR, request objects now pass through a `WithVersionEncoder` which fails on missing kind errors from https://github.com/kubernetes/kubernetes/blob/b722d017a34b300a2284b890448e5a605f21d01e/staging/src/k8s.io/apimachinery/pkg/runtime/scheme.go#L246-L253.